src/net.cpp

```diff
@@ -1377,6 +1378,58 @@ void CConnman::SocketEvents(std::set<SOCKET> &recv_set, std::set<SOCKET> &send_s
         if(FD_ISSET(hSocket, &fdsetError))
             error_set.insert(hSocket);
 }
+#else
+void CConnman::SocketEvents(std::set<SOCKET> &recv_set, std::set<SOCKET> &send_set, std::set<SOCKET> &error_set)
+{
+    std::set<SOCKET> recv_select_set, send_select_set, error_select_set;
+    GenerateSelectSet(recv_select_set, send_select_set, error_select_set);
```

Comment:
Could be worth having `GenerateSelectSet` return a bool that indicates whether or not the sets have any content so that we can avoid the `.empty()` checks below (and the repetition necessary for the legacy `select` impl).